### ISSUE TITLE
Syntax error for passing kwargs to a function inside fstrings

### ISSUE THREAD

**Issue by TotallyNotChase** [microsoft/pylance-release#45]
## Environment data

-   Language Server version: 2020.6.1
-   OS and version: Microsoft Windows [Version 10.0.19041.329]
-   Python version (& distribution if applicable, e.g. Anaconda): Python 3.7.4

## Expected behaviour

Passing a dictionary to a function inside an fstring should not show any syntax errors.

## Actual behaviour

Pylance shows syntax error when encountering a dict (`{}`) inside an `fstring`

## Logs

Doubt the logs would be useful, but here they are anyway-

```
[Info  - 5:07:38 PM] Pylance language server 2020.6.1 starting
[Info  - 5:07:38 PM] Server root directory: c:\Users\Chase\.vscode\extensions\ms-python.vscode-pylance-2020.6.1\server
[Info  - 5:07:38 PM] No configuration file found.
[Info  - 5:07:38 PM] Setting pythonPath for service "test": "C:\Users\Chase\AppData\Local\Programs\Python\Python37-32\python.exe"
[Info  - 5:07:39 PM] stubPath c:\Users\Chase\Desktop\Python Projects\test\typings is not a valid directory.
[Info  - 5:07:39 PM] Assuming Python version 3.7
[Info  - 5:07:39 PM] Assuming Python platform Windows
[Info  - 5:07:39 PM] Searching for source files
[Info  - 5:07:39 PM] Found 1 source file
[Info  - 5:07:39 PM] Background analysis root directory: c:\Users\Chase\.vscode\extensions\ms-python.vscode-pylance-2020.6.1\server
[Info  - 5:07:39 PM] Background analysis started
[FG] parsing: c:\Users\Chase\Desktop\Python Projects\test\test.py (31ms)
[FG] parsing: c:\Users\Chase\.vscode\extensions\ms-python.vscode-pylance-2020.6.1\server\typeshed-fallback\stdlib\2and3\builtins.pyi (377ms)
[FG] binding: c:\Users\Chase\.vscode\extensions\ms-python.vscode-pylance-2020.6.1\server\typeshed-fallback\stdlib\2and3\builtins.pyi (156ms)
[FG] binding: c:\Users\Chase\Desktop\Python Projects\test\test.py (2ms)
[BG] analyzing: c:\Users\Chase\Desktop\Python Projects\test\test.py ...
[BG]   parsing: c:\Users\Chase\Desktop\Python Projects\test\test.py (169ms)
[BG]   parsing: c:\Users\Chase\.vscode\extensions\ms-python.vscode-pylance-2020.6.1\server\typeshed-fallback\stdlib\2and3\builtins.pyi (380ms)
[BG]   binding: c:\Users\Chase\.vscode\extensions\ms-python.vscode-pylance-2020.6.1\server\typeshed-fallback\stdlib\2and3\builtins.pyi (146ms)
[BG]   binding: c:\Users\Chase\Desktop\Python Projects\test\test.py (1ms)
[BG]   checking: c:\Users\Chase\Desktop\Python Projects\test\test.py ...
[BG]     parsing: c:\Users\Chase\.vscode\extensions\ms-python.vscode-pylance-2020.6.1\server\typeshed-fallback\stdlib\3\typing.pyi (78ms)
[BG]     binding: c:\Users\Chase\.vscode\extensions\ms-python.vscode-pylance-2020.6.1\server\typeshed-fallback\stdlib\3\typing.pyi (39ms)
[BG]     parsing: c:\Users\Chase\.vscode\extensions\ms-python.vscode-pylance-2020.6.1\server\typeshed-fallback\stdlib\2and3\_typeshed\__init__.pyi (18ms)
[BG]     binding: c:\Users\Chase\.vscode\extensions\ms-python.vscode-pylance-2020.6.1\server\typeshed-fallback\stdlib\2and3\_typeshed\__init__.pyi (3ms)
[BG]   checking: c:\Users\Chase\Desktop\Python Projects\test\test.py (187ms)
[BG] analyzing: c:\Users\Chase\Desktop\Python Projects\test\test.py (890ms)
[FG] parsing: c:\Users\Chase\.vscode\extensions\ms-python.vscode-pylance-2020.6.1\server\typeshed-fallback\stdlib\3\typing.pyi (68ms)
[FG] binding: c:\Users\Chase\.vscode\extensions\ms-python.vscode-pylance-2020.6.1\server\typeshed-fallback\stdlib\3\typing.pyi (36ms)
[FG] parsing: c:\Users\Chase\.vscode\extensions\ms-python.vscode-pylance-2020.6.1\server\typeshed-fallback\stdlib\2and3\_typeshed\__init__.pyi (23ms)
[FG] binding: c:\Users\Chase\.vscode\extensions\ms-python.vscode-pylance-2020.6.1\server\typeshed-fallback\stdlib\2and3\_typeshed\__init__.pyi (8ms)
```

## Code Snippet / Additional information

```python
from typing import Dict

def foo(spam, dictval: Dict):
    print(dictval)
    return "Done"

print(f"{foo(0, {'bar' : 1, 'baz': 2})}")
```

Output after `python ./test.py` in the commandline-
```powershell
{'bar': 1, 'baz': 2}
Done
```

The code doesn't actually cause a syntax error as you can see. But Pylance reports syntax error-

![image](https://user-images.githubusercontent.com/44284917/86458524-90586680-bd4f-11ea-8c2c-8d3011c1fa9b.png)

It also shows a parameter missing error, however that's certainly a side effect of the syntax error-

![image](https://user-images.githubusercontent.com/44284917/86458637-c4338c00-bd4f-11ea-80a9-d7c5abe14b1c.png)

**Comment by erictraut** [comment]
Thanks for the bug report. Pylance was not correctly handling nested braces within an f-string expression. It is now working. The fix will be in an upcoming release of Pylance.

**Comment by jakebailey** [comment]
This issue has been fixed in version 2020.7.0, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/master/CHANGELOG.md#202070-9-july-2020
